langchain/callbacks/whylabs_callback.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from langchain.callbacks.base import BaseCallbackHandler
+from langchain.schema import AgentAction, AgentFinish, Generation, LLMResult
+
+logger = logging.getLogger(__name__)
+
+
+def import_langkit(
+    sentiment: Optional[bool] = None,
+    toxicity: Optional[bool] = None,
+    themes: Optional[bool] = None,
+) -> Any:
+    try:
+        import langkit  # noqa: F401
+        import langkit.regexes  # noqa: F401
+        import langkit.textstat  # noqa: F401
+
+        if sentiment:
+            import langkit.sentiment  # noqa: F401
+        if toxicity:
+            import langkit.toxicity  # noqa: F401
+        if themes:
+            import langkit.themes  # noqa: F401
+    except ImportError:
+        raise ImportError(
+            "To use the whylabs callback manager you need to have the `langkit` python "
+            "package installed. Please install it with `pip install langkit`."
+        )
+    return langkit
+
+
+class WhyLabsCallbackHandler(BaseCallbackHandler):
+    """WhyLabs CallbackHandler."""
+
+    def __init__(
```

Comment:
will let you decide on the best interface here, but we've found it's often nice to pass instance variable in directly to constructor (in this case just the logger) and then to have a convenience classmethod called something like `from_params` that takes in all the individual params and assembles logger and passes it in. that way you get both a convenient constructor if you want, and if somebody wants to fully control how logger is constructed and pass that in themselves, they can do that, too. also makes things a bit more robust to future changes e.g. if logger interface changes, from_params will be out-of-sync but user can still construct logger themselves and pass it in

Reply:
Thanks, that does sound better! I will make that change within the hour, and then resolve this comment.